receiver/postgresqlreceiver/config.go

```diff
@@ -0,0 +1,98 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package postgresqlreceiver
+
+import (
+	"errors"
+	"fmt"
+
+	"go.opentelemetry.io/collector/receiver/scraperhelper"
+	"go.uber.org/multierr"
+)
+
+type Config struct {
+	scraperhelper.ScraperControllerSettings `mapstructure:",squash"`
+	Username                                string   `mapstructure:"username"`
+	Password                                string   `mapstructure:"password"`
+	Databases                               []string `mapstructure:"databases"`
+	Host                                    string   `mapstructure:"host"`
+	Port                                    int      `mapstructure:"port"`
+	SSLConfig                               `mapstructure:",squash"`
+}
+
+type SSLConfig struct {
+	SSLMode     string `mapstructure:"ssl_mode"`
+	SSLRootCert string `mapstructure:"ssl_root_cert"`
+	SSLCert     string `mapstructure:"ssl_cert"`
+	SSLKey      string `mapstructure:"ssl_key"`
+}
+
+func (c *SSLConfig) Validate() []error {
+	var errs []error
+	validValues := map[string]struct{}{
+		"require":     {},
+		"verify-ca":   {},
+		"verify-full": {},
+		"disable":     {},
+	}
+
+	if _, ok := validValues[c.SSLMode]; !ok {
+		errs = append(errs,
+			fmt.Errorf("SSL Mode '%s' not supported, valid values are 'require', 'verify-ca', 'verify-full', 'disable'. The default is 'require'", c.SSLMode))
+	}
+
+	return errs
+}
+
+// ConnString provides SSL configuration to be used in the connection string
+func (c *SSLConfig) ConnString() string {
```

Comment:
Assuming we switch to `configtls`, this kind of "building" functionality should probably be elsewhere, maybe in a `client.go`.